src/Model/AbstractParking.php

```diff
@@ -0,0 +1,53 @@
+<?php 
+
+namespace App\Model;
+
+use App\ValueObject\Money;
+
+abstract class AbstractParking implements ParkingInterface
+{
+    protected Money $pricePerHour;
+    protected int $numberOfPlaces;
+    protected int $freePlaces;
+
+    public function __construct(Money $pricePerHour, int $numberOfPlaces, int $freePlaces)
+    {
+        $this->pricePerHour   = $pricePerHour;
+        $this->numberOfPlaces = $numberOfPlaces;
+        $this->freePlaces     = $freePlaces;
+    }
+
+    protected abstract function canPark(VehicleInterface $vehicle);
+
+    public function getFreePlaces(): int 
+    {
+        return $this->freePlaces;
+    }
+
+    public function park(VehicleInterface $vehicle): void
+    {
+        if (!$this->hasFreePlaces($vehicle)) {
+            throw new \Exception("This vehicle can not park here!");
+        } 
+        $this->freePlaces--;
+    }
+
+    public function getPrice(\DateTime $parkingStartDate, \DateTime $parkingEndDate): Money
+    {
+        $interval = $parkingStartDate->diff($parkingEndDate);
+        $hours = $interval->h;
+        $min = $interval->i;
+        if ($min > 0) {
+            $hours++;
+        }
+
+        // TO DO
+        return new Money(1);
+    }
+
+    public function hasFreePlaces(): bool
+    {
+        $total = ($this->getFreePlaces()) ?? false;
+        return $total;
```

Comment:
Nie bardzo rozumiem ten zapis.

nie łatwiej zrobić po prostu return $this->freePlaces > 0;?